### PR TITLE
feat: avoid string-copy in insert-values

### PR DESCRIPTION
**Pull request by ygf11**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
1. For `insert-values`, parser should skip value-to-expr parsing. (Need to support in sqlparser)
2. DfParser pass the values-slice to analyzer.
3. `analyzer` do valuseSource-parse or fallback to expression-executor. 

## Changelog

- Improvement
- Performance Improvement

## Related Issues

Related #4413 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/sql/parsers/parser_insert.rs [hunk 1]
```diff
@@ -38,19 +66,92 @@ impl<'a> DfParser<'a> {
...
+                    table,
+                    on,
+                }))
+            }
             _ => parser_err!("Expect set insert statement"),
         }
     }
+
+    fn raw_values_str(&self, table_name: &ObjectName, table: bool) -> Result<&'a str, ErrorCode> {
+        let mut reader = BufReader::new(Cursor::new(self.sql));
```
[the remark this answers is not in the document]
Comment (ygf11) [reply query/src/sql/parsers/parser_insert.rs:103]: Yes, current way is rough. 

I am thinking about add some info in parser, like:
```
   values-token -> position in sql 
```

Then we can direct get the values slice.

[the remark this answers is not in the document]
Comment (sundy-li) [reply query/src/sql/parsers/parser_insert.rs:103]: I think it's reasonable to add lifetime in the Statement.

But I do not think it's good to parse the `insert values` inside the parser.
So, let's don't parse the values and just return `Option<&str>` to represent the substring in the parser.


And I will parser this `&str` into `Blocks` inside the `source_values` in another pr.

[the remark this answers is not in the document]
Comment (ygf11) [reply query/src/sql/parsers/parser_insert.rs:103]: Parse `insert values` in source_values is ok.

But I worry about that we need support running multiple statements.
For single statement, start of the query is also the start of the statement.
But for multiple statement, we need to know the start of every-insert-values statement, which need by the `source_values`.

[the remark this answers is not in the document]
Comment (sundy-li) [reply query/src/sql/parsers/parser_insert.rs:103]: In the sqlparser,  if we already parse the `insert into xx values`, then we should move the token until reach EOF or next statement.

[the remark this answers is not in the document]
Comment (ygf11) [reply query/src/sql/parsers/parser_insert.rs:103]: If I miss some contexts, please point it out:)

For example, we have a query with two statements.
`select * from t; insert into t values(1),(2),(3),(4);`

So we will get two statements.
1. `select * from t`
2. `insert into t values(1),(2),(3),(4);`

For the second statement, we need to give `inser sql` to `source_values`, but we do not know the start in the total query, we only hold the total query.
```
insert sql: insert into t values(1),(2),(3),(4);
total query: select * from t; insert into t values(1),(2),(3),(4);
```

The reason is sqlparser does not save mapping from token to it's position in original query.

[the remark this answers is not in the document]
Comment (sundy-li) [reply query/src/sql/parsers/parser_insert.rs:103]: If we return `&str`, it's a string slice of your Owner String. So everything is ok now?

[the remark this answers is not in the document]
Comment (ygf11) [reply query/src/sql/parsers/parser_insert.rs:103]: Ok.

**Comment by ygf11** [comment]
close, move to #4730.

**Pull request closed without merging**